src/actions/Bond.actions.js

```diff
@@ -235,16 +237,7 @@ export const bondAsset =
       // TODO: it may make more sense to only have it in the finally.
       // UX preference (show pending after txn complete or after balance updated)
 
-      const reserveContract = contractForReserve({ bond, provider, networkID });
-
-      if (bond === BONDS.ohm_dai || bond === BONDS.ohm_frax) {
-        balance = await reserveContract.balanceOf(address);
-      } else if (bond === BONDS.dai) {
-        balance = await reserveContract.balanceOf(address);
-        balance = ethers.utils.formatEther(balance);
-      }
-
-      return dispatch(fetchBondSuccess({ bond, balance }));
+      return dispatch(calculateUserBondDetails({ address, bond, networkID, provider }));
```

Comment:
Why did you remove the lines above?

Reply:
because it is being executed in `calculateUserBondDetails`.
it's better to just have all that logic in `calculateUserBondDetails` anyways so that in the future when we add more bonds and whatnot, we only have to edit it in one place.